### PR TITLE
Add documentation build process using Sphinx

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

New Features:
- Added documentation build process using Sphinx.
- Removed hugging face token input as its legacy and not used. Commented out in case we need to come back to this approach (switch LLM engines)
- Moved generate API key button to be inline with text box (looks cleaner)

Before:
image

After:
image

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Toolbox Installer'
+copyright = '2024, ClinicianFOCUS'
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply docs/conf.py:10]: This is okay its in the sphinx stuff.

File: docs/how_to_install.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+         - **FreeScribe**
+
+      Recommended Actions:
+         - Start Docker Desktop before launching Local LLM and Speech2Text.
+         - Launch Local LLM and Speech2Text to build the container images.
+         - Wait for the build process to complete.
+
+5. **Verify Installation**
+
+      Verify that all installed components are running correctly. Ensure Docker Desktop is running and the necessary containers are active. If error occurs you will encounter installation errors.
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply docs/how_to_install.rst:110]: No. the install logs dont overly reveal anythin ill reword.

[the remark this answers is not in the document]
Comment (ItsSimko) [reply docs/how_to_install.rst]: https://github.com/ClinicianFOCUS/clinicianfocus-installer/pull/52/commits/06524a6d44d12d2a90118425869e83bf2e52e9fd

File: docs/how_to_install.rst [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+1. **API Key Setup:** Enter or generate a secure API key for service access.
+2. **Model Selection:** Choose a pre-configured or custom model, and provide your Hugging Face API token if required.
+3. **Whisper Model Selection:** Select a model size based on your resource availability and accuracy requirements.
+4. **Finalize Installation:** Click **Install** to complete the setup.
+
+Additional Notes
+^^^^^^^^^^^^^^^^
+
+- The installation requires sufficient disk space for the selected models.
+- Ensure internet connectivity for downloading model.
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply docs/how_to_install.rst]: https://github.com/ClinicianFOCUS/clinicianfocus-installer/pull/52/commits/a5fb0c12d192ccf9869285a7e627380a92152f27

**Review by yingbull — APPROVED** [review]

**Pull request merged**